File: leetcode/java/709. To Lower Case.java
class Solution {
    public String toLowerCase(String str) {
        final char[] array = new char[str.length()];
        final int size = str.length();
        
        for(int i = 0; i < size; i++) {
            final char c = str.charAt(i);
            array[i] = c;
            
            if(array[i] >= 'A' && array[i] <= 'Z') {
                array[i] += 32;
            }
        }
        
        return new String(array);
    }
}

File: leetcode/kotlin/709. To Lower Case.kt
class Solution {
    fun toLowerCase(str: String): String {
        val array = CharArray(str.length)
        var index = 0
        
        for(element in str) {
            if(element >= 'A' && element <= 'Z') {
                array[index++] = (element + 32)
                continue
            }
            
            array[index++] = element
        }
        
        return String(array)
    }
}
